expression/builtin_cast.go

```diff
@@ -1011,6 +1011,38 @@ type builtinCastDecimalAsRealSig struct {
 	baseBuiltinCastFunc
 }
 
+func (b *builtinCastDecimalAsRealSig) PropagateType() {
```

Comment:
Can we use normal `func ProgateRealType(arg Expression)` to do this? So we can call the function in other builtinFunction(eg. `builtinCastStringAsRealSig`) in future?